lib/node/runner/isaac_state_manager.go

```diff
@@ -16,13 +16,14 @@ import (
 type ISAACStateManager struct {
 	sync.RWMutex
 
-	nr              *NodeRunner
-	state           consensus.ISAACState
-	stateTransit    chan consensus.ISAACState
-	stop            chan struct{}
-	blockTimeBuffer time.Duration              // the time to wait to adjust the block creation time.
-	transitSignal   func(consensus.ISAACState) // the function is called when the ISAACState is changed.
-	genesis         time.Time                  // the time at which the GenesisBlock was saved. It is used for calculating `blockTimeBuffer`.
+	nr                           *NodeRunner
+	state                        consensus.ISAACState
+	stateTransit                 chan consensus.ISAACState
+	stop                         chan struct{}
+	blockTimeBuffer              time.Duration              // the time to wait to adjust the block creation time.
+	transitSignal                func(consensus.ISAACState) // the function is called when the ISAACState is changed.
+	firstConsensusBlockTime      time.Time                  // the time at which the first consensus block was saved(height 2). It is used for calculating `blockTimeBuffer`.
+	isSetFirstConsensusBlockTime bool                       // the flag for checking firstConsensusBlockTime is set. It is helpful for performance.
```

Comment:
How is it useful for performance ?

Reply:
It's because when this field is assigned true, no more assign `firstConsensusBlockTime`.
And I removed this field because of this [good idea](https://github.com/bosnet/sebak/pull/661#discussion_r231385271) :)